src/instrumentations/xhr.ts

```diff
@@ -727,7 +729,24 @@ export function instrumentXHR(config: XhrConfig) {
     }
   }
 
+  if (config.networkHeadersCapture) {
+    XMLHttpRequest.prototype.setRequestHeader = function (
+      this: XMLHttpRequest,
+      ...args
+    ) {
+      const [key, value] = args;
+      const xhrMem = _xhrMem.get(this);
+      if (xhrMem && key && value) {
+        const newKey = `http.request.header.${key.toLowerCase()}`;
```

Comment:
Let's try to make sure the headers extraction matches what otel does in their http instrumentation (https://github.com/open-telemetry/opentelemetry-js/blob/82b7526b028a34a23936016768f37df05effcd59/experimental/packages/opentelemetry-instrumentation-http/src/utils.ts#L583)

Reply:
updated!